### DISCUSSION TITLE
Format of Function call when a specific function is required?

### DISCUSSION THREAD

**Discussion by StefanBristol** [MacPaw/OpenAI#90] [Q&A]
Hi, I've got the function call working in the query, using the default function_call = "auto".
i.e.  let query = ChatQuery(model: .gpt4, messages: xxxxxxxx, functions: xxxxxxx)

But I can't seem to work out from the code what the format is for....
let query = ChatQuery(model: .gpt4, messages: xxxxxx, functions: xxxxxxx, functionCall: ???)

For the example, it is a function called 'getWidget' and will have properties 'type' and 'name'.

Any help on how to use the OpenAI package in this case, would be most welcome.

**Comment by kalafus** [comment]
Hi @StefanBristol ! Any progress?
